wordpress-mobile/WordPress-iOS
# Adds Domains Dashboard to My Site (and a Feature Flag for it)

Fixes #17067, #17068, #17069

This PR adds a Feature Flag for Domains, and an entry point and initial implementation for the Domains Dashboard, accessible from My Site


To test:
Pre requisites:
- Use a store sandbox token to avoid actual domain purchases
- Have at least one site with a free plan
- Have at least one site with a paid plan that has not redeemed the free domain yet

Testing steps:
Test on iPhone and iPad
- Build/run and go to My Site
- Make sure the "Domains" entry appears in the "Configure" section (see screenshot)
- Select a site with a paid plan that has not yet redeemed the domain yet
- Tap on "Domains" and make sure the Domains Dashboard appears and looks like in the screenshot below (again, the duplicate "Register a domain" entry will be removed when it's final)
- Tap on "Register a free domain" and complete the flow: make sure the domain registration completes successfully, and that you see the registration epilogue at the end.
- Make sure that after dismissing the epilogue screen, you are redirected to the dashboard, that you don't see the "Register a free domain" section anymore and that the primary domain has been updated accordingly
- Go back to my site and select a free plan
- Tap on "Domains" and make sure you don't see the "Register a free domain" section

<p align=center>
<img width="282" src="https://user-images.githubusercontent.com/34376330/130532011-f8f07ed5-5139-4c87-b819-c3f740cc6996.png">
</p>


## Regression Notes
1. Potential unintended areas of impact
None: this buds a brand new view

2. What I did to test those areas of impact (or what existing automated tests I relied on)


3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
